lib/core/test/unit/Cardano/Wallet/Primitive/AddressDerivation/RandomSpec.hs

```diff
@@ -66,24 +72,57 @@ prop_keyDerivation seed encPwd ix1 ix2 =
                                   Golden tests
 -------------------------------------------------------------------------------}
 
+goldenSpec :: Spec
+goldenSpec = describe "Golden tests" $ do
+    it "generateKeyFromSeed - no passphrase" $
+        generateTest test1
+
+    it "generateKeyFromSeed - with passphrase" $
+        generateTest test2
+
 data GenerateKeyFromSeed = GenerateKeyFromSeed
-    { seed :: Passphrase "seed"
+    { mnem :: [Text]
     , pwd :: Passphrase "encryption"
     , rootKey :: Key 'RootK XPrv
     } deriving (Show, Eq)
 
 generateTest :: GenerateKeyFromSeed -> Expectation
 generateTest GenerateKeyFromSeed{..} =
-    generateKeyFromSeed seed pwd `shouldBe` rootKey
+    generateKeyFromSeed (Passphrase seed) pwd `shouldBe` rootKey
+  where
+    Right (Passphrase seed) = fromMnemonic @'[12] mnem
 
 test1 :: GenerateKeyFromSeed
 test1 = GenerateKeyFromSeed
-    { seed = pp "df3fecb0196b9ef71fae1f6db886954d4ff9914fa075c665de03f17ca698398a"
+    { mnem = defMnem
+    , pwd = pp ""
+    , rootKey = xprv16 "b84d0b6db447911a98a3ade98145c0e8323e106f07bc17a99c2104c2688bb7528310902a3cec7e262ded6a4369ec1f48966a6b48b1ee90aa00e61b95417949f81258854ab44b0cfda59bd68fbd87f280841a390068049df0f8a903c94ba65b7aa4762129a6c83acfda5b257eaeb73ec5fee1518b6674fdc7891fe23f06174421"
+    }
+
+test2 :: GenerateKeyFromSeed
+test2 = GenerateKeyFromSeed
+    { mnem = defMnem
     , pwd = pp "4a87f05fe25a57c96ff5221863e61b91bcca566b853b616f55e5d2c18caa1a4c"
-    -- fixme: this was not generated by the previous wallet code.
-    , rootKey = xprv16 "8001a61113793d1f90a78eb49a5a12c49a3a606629bcde236cdd6724478245fdd671815607e8ae10491e68db54f4322ad3723e7de0d06a90754ac5cf581184fb9471cfd860d3d28af4e025eaee37faf2b5493822afee2002ae7eabfbb7f579c604da83ffe99c45e3e4a86e4914a75d98d6b5ad9f7c7b68e5836b02a1377fedc5"
+    , rootKey = xprv16 "c7afbf0c46212c44c5b6a10cf2e7e0a94140f792b8f58964b17a7a67b761619aeed21598a7dde167e0e34f05b5b72c9b61248f4374d858751482f154eea106511258854ab44b0cfda59bd68fbd87f280841a390068049df0f8a903c94ba65b7aa4762129a6c83acfda5b257eaeb73ec5fee1518b6674fdc7891fe23f06174421"
     }
 
+{- test data was generated with the following:
+genKey :: ByteString -> ByteString
+genKey spendingPassword = convertToBase Base16 (unXPrv prv)
+  where
+    spendingPassword' = PassPhrase pp
+    Right pp = convertFromBase Base16 spendingPassword
+    mnemonic = def :: Mnemonic 12
+    (_, EncryptedSecretKey prv _hash) =
+        safeDeterministicKeyGen seed spendingPassword'
+    seed = Mnemonic.mnemonicToSeed mnemonic
+-}
+
+-- | This is the mnemonic that provides the 'Default' instance in cardano-sl
+defMnem :: [Text]
+defMnem = [ "squirrel", "material", "silly", "twice", "direct", "slush"
+          , "pistol", "razor", "become", "junk", "kingdom", "flee" ]
```

Comment:
Perhaps `defMnemonic` or `defaultMnemonic` would be more readable here?

I can totally understand using short names for values used internally, but since this is a top-level value, perhaps we can afford to use a more readable name? (I mean, we're not running out of space here yet :))

Reply:
Fixed, thanks.